### PR TITLE
LF

### PR DESCRIPTION
**Pull request by LenaTheWolfess**

### CODE REVIEW

File: simulation/components/GarrisonHolder.js [hunk 1]
```diff
@@ -45,12 +45,14 @@ GarrisonHolder.prototype.Schema =
...
 							"<data type='decimal'/>" +
 						"</element>" +
+						"<optional>" +
 						"<element name='VisibleGarrisonAllowedClasses'>" +
 							"<attribute name='datatype'>" +
 								"<value>tokens</value>" +
 							"</attribute>" +
 							"<text/>" +
 						"</element>" +
+						"</optional>"+
```
Comment (StanleySweet) [inline simulation/components/GarrisonHolder.js:55]: Fix indent

File: simulation/components/GarrisonHolder.js [hunk 2]
```diff
@@ -77,17 +79,18 @@ GarrisonHolder.prototype.Init = function()
 	if (this.template.VisibleGarrisonPoints)
 	{
 		let points = this.template.VisibleGarrisonPoints;
-		for (let point in points)
+		for (let point in points) {
```
Comment (StanleySweet) [inline simulation/components/GarrisonHolder.js:82]: Coding conventions (I'm guessing auto format)

File: simulation/components/GarrisonHolder.js [hunk 3]
```diff
@@ -77,17 +79,18 @@ GarrisonHolder.prototype.Init = function()
...
 			this.visibleGarrisonPoints.push({
 				"offset": {
 					"x": +points[point].X,
 					"y": +points[point].Y,
 					"z": +points[point].Z,
-					"allowedClasses": points[point].VisibleGarrisonAllowedClasses
+					"allowedClasses": points[point].VisibleGarrisonAllowedClasses ? points[point].VisibleGarrisonAllowedClasses : "Infantry+Ranged"
 				},
```
Comment (StanleySweet) [inline simulation/components/GarrisonHolder.js:89]: If you go that road templates should be updated as well

File: simulation/components/GarrisonHolder.js [hunk 4]
```diff
@@ -185,13 +188,22 @@ GarrisonHolder.prototype.IsAllowedToGarrison = function(ent)
...
 };
 
+/**
+* Return string of allowed classes in the form "Class1 Class2+Class3" from visibleGarrisonPoint to handle default value
+*/
+GarrisonHolder.prototype.GetVisibleAllowedClasses = function (visibleGarrisonPoint)
+{
+	if (!visibleGarrisonPoint)
+		return "";
+	return visibleGarrisonPoint.offset.allowedClasses._string || visibleGarrisonPoint.offset.allowedClasses;
```
Comment (StanleySweet) [inline simulation/components/GarrisonHolder.js:198]: Why is that double check needed ? Is there a case where one of them are  null

Comment (LenaTheWolfess) [reply simulation/components/GarrisonHolder.js:198]: If I use "Infantry+Ranged" above, _string is not defined, the second part is there just for that one case

File: simulation/components/GarrisonHolder.js (hunk 1 above)
Comment (LenaTheWolfess) [reply simulation/components/GarrisonHolder.js:55]: did

File: simulation/components/GarrisonHolder.js (hunk 2 above)
Comment (LenaTheWolfess) [reply simulation/components/GarrisonHolder.js:82]: removed

File: simulation/components/GarrisonHolder.js (hunk 3 above)
Comment (LenaTheWolfess) [reply simulation/components/GarrisonHolder.js:89]: I am not sure what do you mean.

**Pull request closed without merging**